src/nvim/screen.c

```diff
@@ -1601,6 +1601,9 @@ static void win_update(win_T *wp, DecorProviders *providers)
    * End of loop over all window lines.
    */
 
+  if (wp->w_p_rnu && buf == curwin->w_buffer) {
+    wp->w_old_cursor_lnum_rnu = curwin->w_cursor.lnum;
```

Comment:
This causes `'relativenumber'` not to be redrawn with splits and `nvim_win_set_cursor()`:
1. Open two splits with the same buffer
2. Move the cursor to a different line in one window
3. Use `nvim_win_set_cursor()` to move the cursor in the other window to a different line
4. Use `nvim_win_set_cursor()` to move the cursor in the other window to the same line

Reply:
Using `curwin` here doesn't seem to make sense.